### PR TITLE
Shape HeadersInit from FetchParameters

### PR DESCRIPTION
**Pull request by freaz**
## Description

Added function to format Record to array of tuples which is expected in `HeadersInit` and support duplicate headers.

The original test is indented, so it looks long. So, check it with [hidden whitespaces](https://github.com/superfaceai/one-sdk-js/pull/293/files?diff=split&w=1).

## Motivation and Context

TypeScript compiler was complaining about the shape of `headersInit` data.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](https://github.com/superfaceai/one-sdk-js/blob/main/SECURITY.md) is correct.
- [x] I have read the [CONTRIBUTING](https://github.com/superfaceai/one-sdk-js/blob/main/CONTRIBUTING.md) document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/node/fetch/fetch.node.ts [hunk 1]
```diff
@@ -239,4 +234,22 @@ export class NodeFetch implements IFetch, Interceptable, AuthCache {
...
+
+    const headers: [string, string][] = [];
+
+    Object.entries(data).forEach(([key, value]) => {
+      if (Array.isArray(value)) {
+        value.forEach((val) => headers.push([key, val]));
+      } else {
+        headers.push([key, value]);
+      }
+    });
```
Comment (jnv) [inline src/node/fetch/fetch.node.ts:251]: thought: Would using `flatMap` make this more readable?

```suggestion
    headers = Object.entries(data).flatMap(([key, value]) => {
      if (Array.isArray(value)) {
        return value.map((nestedVal) => [key, nestedVal])
      }
      return [[key, value]]
    }
```

But it's probably less performant (but so is `forEach`).

Comment (freaz) [reply src/node/fetch/fetch.node.ts:251]: Thank you, but I intentionally implemented this way.

File: src/node/fetch/fetch.node.test.ts [hunk 2]
```diff
@@ -15,225 +15,292 @@ const timers = new MockTimers();
...
+      const binaryContentTypes = [
+        'application/octet-stream',
+        'audio/mp3',
+        'audio/wav',
+        'audio/wav;rate=8000',
+        'video/mp4',
+        'image/jpeg',
+      ];
+
+      for (const contentType of binaryContentTypes) {
```
Comment (TheEdward162) [inline src/node/fetch/fetch.node.test.ts:215]: question: any idea why this doesn't use https://jestjs.io/docs/api#testeachtablename-fn-timeout

File: src/node/fetch/fetch.node.ts (hunk 1 above)
Comment (TheEdward162) [reply src/node/fetch/fetch.node.ts:251]: I think at that point I would find a classic `for .. of` loop more readable, but I think this code is short enough to be fine either way.

**Review by TheEdward162 — APPROVED** [review]

File: src/node/fetch/fetch.node.test.ts (hunk 2 above)
Comment (freaz) [reply src/node/fetch/fetch.node.test.ts:215]: Don't know. But I just rewrote when I changed `content-type` handling in https://github.com/superfaceai/one-sdk-js/blob/dev/src/core/interpreter/http/filters.ts#L259

Comment (freaz) [reply src/node/fetch/fetch.node.test.ts:215]: Now (when I ran the test) I know why. `beforeEach` can't be used inside `test.each`.

**Pull request merged**